x/dymns/keeper/hooks.go

```diff
@@ -0,0 +1,370 @@
+			)
+			// ignore the invalid entries for now, invariant will catch it
+			continue
+		}
+
+		finishedSellOrders = append(finishedSellOrders, *so)
+	}
+
+	return
+}
+
+/* -------------------------------------------------------------------------- */
+/*                             x/rollapp hooks                                */
+/* -------------------------------------------------------------------------- */
+
+// GetRollAppHooks returns the RollApp hooks struct.
+func (k Keeper) GetRollAppHooks() rollapptypes.RollappHooks {
+	return rollappHooks{
+		Keeper: k,
+	}
+}
+
+type rollappHooks struct {
+	Keeper
+}
+
+var _ rollapptypes.RollappHooks = rollappHooks{}
+
+func (h rollappHooks) RollappCreated(ctx sdk.Context, rollappID, alias string, creatorAddr sdk.AccAddress) error {
+	if alias == "" {
+		return nil
+	}
+
+	// ensure RollApp record is set
+	if !h.Keeper.IsRollAppId(ctx, rollappID) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "not a RollApp chain-id: %s", rollappID)
+	}
+
+	if !dymnsutils.IsValidAlias(alias) {
+		return errorsmod.Wrapf(gerrc.ErrInvalidArgument, "invalid alias format: %s", alias)
+	}
+
+	if !h.Keeper.CanUseAliasForNewRegistration(ctx, alias) {
+		return errorsmod.Wrapf(gerrc.ErrAlreadyExists, "alias already in use or preserved: %s", alias)
+	}
+
+	priceParams := h.Keeper.PriceParams(ctx)
+
+	aliasCost := sdk.NewCoins(
+		sdk.NewCoin(
+			priceParams.PriceDenom, priceParams.GetAliasPrice(alias),
+		),
+	)
+
+	return h.Keeper.registerAliasForRollApp(ctx, rollappID, creatorAddr, alias, aliasCost)
+}
+
+func (h rollappHooks) BeforeUpdateState(_ sdk.Context, _ string, _ string) error {
+	return nil
+}
+
+func (h rollappHooks) AfterStateFinalized(_ sdk.Context, _ string, _ *rollapptypes.StateInfo) error {
+	return nil
+}
+
+func (h rollappHooks) FraudSubmitted(_ sdk.Context, _ string, _ uint64, _ string) error {
+	return nil
+}
+
+// FutureRollappHooks is temporary added to handle future hooks that not available yet.
+type FutureRollappHooks interface {
+	OnRollAppIdChanged(ctx sdk.Context, previousRollAppId, newRollAppId string) error
+	// Just a pseudo method signature, the actual method signature might be different.
+
+	// TODO DymNS: connect to the actual implementation when the hooks are available.
+	//   The implementation of OnRollAppIdChanged assume that both of the RollApp records are exists in the x/rollapp store.
+}
+
+var _ FutureRollappHooks = rollappHooks{}
+
+func (k Keeper) GetFutureRollAppHooks() FutureRollappHooks {
+	return rollappHooks{
+		Keeper: k,
+	}
+}
+
+func (h rollappHooks) OnRollAppIdChanged(ctx sdk.Context, previousRollAppId, newRollAppId string) error {
+	// This can be call when the RollAppId is changed due to fraud submission,
+	// so due to the critical nature of the reason,
+	// the following execution will be done in branched context to prevent any side effects.
+	// If any error occurs, the state change to this module will be discarded, no error returned to the caller.
```

Comment:
```suggestion
```

Reply:
removed this long and put a shorter desc with just enough information at more accurate place. Thanks